onedal/covariance/incremental_covariance.py

```diff
@@ -86,21 +86,24 @@ def partial_fit(self, X, y=None, queue=None):
         """
         X = _check_array(X, dtype=[np.float64, np.float32], ensure_2d=True)
 
-        if not hasattr(self, "_policy"):
-            self._policy = self._get_policy(queue, X)
+        if not hasattr(self, "_queue"):
+            self._queue = queue
```

Comment:
It is for finalize_fit dispatching. It does not have data argument, so, in case if user does not provide queue explicitly then the last queue from partial_fit is used.

Reply:
Can't the last queue be extracted from the stored policy via the `_queue` property? A lot of this logic might be unnecessary. @olegkkruglov let me know.

Reply:
It was done like that before this PR. but now it turned out that different policies must be used in finalize_fit and partial_fit that's why stored policy from partial fit is not acceptable

Reply:
Since it's just partial_fit and reset which requires a data parallel policy, I would overload partial_fit and reset in the spmd interface, and make sure that the correct policy is taken there via a super call.  A small duplication of code, but at least its clear to the developer and the user on what is going on, and its located in a place that someone looking to understand the spmd interface can see the limitations of partial_fit for spmd in the incremental algos. It would be then more straightforward for @samir-nasibli 's request for a refactor with an Incremental base class.

Reply:
> It is for finalize_fit dispatching. It does not have data argument, so, in case if user does not provide queue explicitly then the last queue from partial_fit is used.

Maybe it make sense explicitly ask user provide sycl queue? Otherwise it is headache

Reply:
finalize_fit is called implicitly on sklearnex side. if we want to keep scikit-like interface (without explicit finalize call) the only option is to call finalize after every call of partial_fit. this option was rejected on arch meeting, that's why keeping queue in attributes is currently unavoidable.